rtw_excel_report/models/models.py

```diff
@@ -1118,55 +1118,28 @@ def _compute_sale_order_name(self):
     def _compute_sale_order_line_name_excel(self):
         for line in self:
-            categ_name=""
-            if line.product_id and line.product_id.product_tmpl_id: 
-                if line.product_id.product_tmpl_id.config_ok :  
-                    if line.product_id.product_tmpl_id.categ_id.name:
-                        categ_name = line.product_id.product_tmpl_id.categ_id.name
-                    elif line.product_id.product_tmpl_id.product_no :
-                        categ_name = line.product_id.product_tmpl_id.product_no
-                    else: 
-                        categ_name = line.product_id.product_tmpl_id.name   
+            categ_name = ""
+            prod = line.product_id
+            prod_tmpl_id = line.product_id.product_tmpl_id
+            if prod and prod_tmpl_id:
+                if prod_tmpl_id.config_ok:  
+                    if prod_tmpl_id.categ_id and prod_tmpl_id.categ_id.name:
+                        categ_name = prod_tmpl_id.categ_id.name
+                    elif prod_tmpl_id.product_no:
+                        categ_name = prod_tmpl_id.product_no
+                    elif prod_tmpl_id.name: 
+                        categ_name = prod_tmpl_id.name   
                 else:
-                    # case product is standard Prod + download payment
-                    if line.product_id.product_tmpl_id.seller_ids and line.order_id.partner_id.id:
-                        matching_sup = None  
-                        for sup in line.product_id.product_tmpl_id.seller_ids:
-                            if sup.name.id == line.order_id.partner_id.id:
-                                matching_sup = sup 
-                                break
-                        if matching_sup:
-                            product_code = ("[" + str(matching_sup.product_code) + "]") if matching_sup.product_code else ''
-                            product_name = str(matching_sup.product_name) if matching_sup.product_name else ''
-                            categ_name = product_code + product_name
-                        else:
-                            categ_name =  line.product_id.product_tmpl_id.name
-                    else:
-                        if line.product_id.product_tmpl_id.default_code:
-                            categ_name = "[" +str(line.product_id.product_tmpl_id.default_code) +"]" + line.product_id.product_tmpl_id.name
-                        else:
-                            categ_name =  line.product_id.product_tmpl_id.name
-                                
+                    if line.name:
+                        categ_name = line.name
+                        
             p_type = ""
             if line.p_type:
-                if line.p_type == "special":
-                    p_type = "別注"
-                elif line.p_type == "custom":
-                    p_type = "特注"
+                p_type = "別注" if line.p_type == "special" else "特注" if line.p_type == "custom" else ""
```

Comment:
với t.hợp nhiều case như này thì e cứ viết if else bình thường ra, đừng dùng toán tử 3 ngôi nữa tránh nhầm lẫn.